esp/services/ws_dali/ws_daliservice.cpp

```diff
@@ -96,3 +239,50 @@ void CWSDaliSoapBindingEx::exportSDSData(CHttpRequest* request, CHttpResponse* r
     io.clear();
     removeFileTraceIfFail(outFileNameWithPath);
 }
+
+int CWSDaliSoapBindingEx::onStartUpload(IEspContext &ctx, CHttpRequest* request, CHttpResponse* response, const char *serv, const char *method)
+{
+    StringArray fileNames, files;
+    VStringBuffer source("WsDali::%s()", method);
+    Owned<IMultiException> me = MakeMultiException(source);
+    try
+    {
+#ifdef _USE_OPENLDAP
+        request->queryContext()->ensureSuperUser(ECLWATCH_SUPER_USER_ACCESS_DENIED, "Access denied, administrators only.");
+#endif
+        if (wsdService->isDaliDetached())
+            throw makeStringException(ECLWATCH_CANNOT_CONNECT_DALI, "Dali detached.");
+
+        if (strieq(method, "Import"))
+        {
+            StringBuffer path, add;
+            request->getParameter("Path", path);
+            request->getParameter("Add", add);
+            if (isEmptyString(path))
+                throw makeStringException(ECLWATCH_INVALID_INPUT, "Data path not specified.");
+
+            request->readContentToFiles(nullptr, daliFolder, fileNames);
+            unsigned count = fileNames.ordinality();
+            if (count == 0)
+                throw MakeStringException(ECLWATCH_INVALID_INPUT, "Failed to read upload content.");
+            //For now, we only support importing 1 file per Import request for a better response time.
```

Comment:
is this comment relevant? would be ever want>1 files uploaded at this point?

Reply:
Will remove.